### PR TITLE
Display correct error message when passing Cluster based resource without ClusterClass as part of cluster create

### LINKED ISSUE #3054: Add validation and throw proper error when Cluster resource without ClusterClass is passed with cluster create command
**Bug description**
Currently, when user passes Cluster resource as part of `tanzu cluster create --file <input.yaml>` and the cluster resource doesn't include ClusterClass as part of `.spec.topology.cluster` a wrong error message is thrown which can be misleading to the user.

As part of the error message, throw meaningful error message as this usecase is not yet supported.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**
Should throw meaningful error message as this usecase is not yet supported.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Display correct error message when passing Cluster based resource without ClusterClass as part of cluster create

Currently, when user passes `Cluster` resource as part of `tanzu cluster create --file <input.yaml>` and the cluster resource doesn't include ClusterClass as part of `.spec.topology.cluster` thrown proper error message.

### Which issue(s) this PR fixes

Fixes #3054

### Describe testing done for PR
- Added unit tests
- Manually verified by passing `--dru-run` output to `tanzu cluster create --file <manifest.yaml>` and verified the error message

### Release note

```release-note
Display correct error message when passing Cluster resource without ClusterClass as part of cluster create
```

### CODE REVIEW

**Review by miclettej — APPROVED** [review]

File: pkg/v1/tkg/constants/clusterclass_mapping.go [hunk 1]
```diff
@@ -279,7 +279,8 @@ const (
 
 	SPEC = "spec"
 
-	TopologyClassIncorrectValueErrMsg = "input cluster class file, attribute spec.topology.class has no value or incorrect value or not following correct naming convension"
+	TopologyClassIncorrectValueErrMsg                = "input cluster class file, attribute spec.topology.class has no value or incorrect value or not following correct naming convension"
```
Comment (miclettej) [inline pkg/v1/tkg/constants/clusterclass_mapping.go:282]: misspelling 'convension' should be 'convention'

Comment (anujc25) [reply pkg/v1/tkg/constants/clusterclass_mapping.go]: Thanks for catching this. fixed.

**Review by tenczar — APPROVED** [review]
LGTM

**Review by prkalle — APPROVED** [review]
LGTM

File: pkg/v1/tkg/constants/clusterclass_mapping.go [hunk 2]
```diff
@@ -279,7 +279,8 @@ const (
 
 	SPEC = "spec"
 
-	TopologyClassIncorrectValueErrMsg = "input cluster class file, attribute spec.topology.class has no value or incorrect value or not following correct naming convension"
+	TopologyClassIncorrectValueErrMsg                = "input cluster class file, attribute spec.topology.class has no value or incorrect value or not following correct naming convention"
+	ClusterResourceWithoutTopologyNotSupportedErrMsg = "Input file contains Cluster resource which doesn't have ClusterClass specified. Passing Cluster resource without ClusterClass specification is not supported"
```
Comment (marckhouzam) [inline pkg/v1/tkg/constants/clusterclass_mapping.go:283]: Nit: the first error message starts with a lower-case (`input`) while the new one start with uppercase (`Input`).

I don't yet know how to trigger the error, so I don't have an opinion on which format is best but I was thinking we should be consistent.

File: pkg/v1/tkg/tkgctl/helpers_test.go [hunk 3]
```diff
@@ -211,3 +212,66 @@ var _ = Describe("Cluster Class - IP Family Validation related test cases: ", fu
...
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: []
+`
+			})
+			It("should return false with error", func() {
+				Expect(isClusterClassBased).To(BeFalse())
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(""))
```
Comment (marckhouzam) [inline pkg/v1/tkg/tkgctl/helpers_test.go:264]: Should the new error be specified in `ContainSubstring()` here?

File: pkg/v1/tkg/tkgctl/helpers_test.go [hunk 4]
```diff
@@ -211,3 +212,66 @@ var _ = Describe("Cluster Class - IP Family Validation related test cases: ", fu
...
+			})
+		})
+		When("File doesn't contain cluster resource", func() {
+			BeforeEach(func() {
+				configFileContent = ``
+			})
+			It("should return false with error", func() {
```
Comment (marckhouzam) [inline pkg/v1/tkg/tkgctl/helpers_test.go:271]: Should the description of the test say "without error"?  That is what the check below seems to expect.

File: pkg/v1/tkg/tkgctl/helpers_test.go (hunk 3 above)
Comment (chandrareddyp) [reply pkg/v1/tkg/tkgctl/helpers_test.go:264]: +1

**Comment by chandrareddyp** [comment]
LGTM

File: pkg/v1/tkg/tkgctl/helpers_test.go (hunk 3 above)
Comment (anujc25) [reply pkg/v1/tkg/tkgctl/helpers_test.go]: Thanks for catching this. Update this to use `ClusterResourceWithoutTopologyNotSupportedErrMsg`

**Comment by marckhouzam** [comment]
LGTM

**Pull request merged**